core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -54,6 +57,8 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
   import org.apache.kafka.clients.consumer.KafkaConsumer
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
+  val offsets = new HashMap[TopicPartition, Long]()
```

Comment:
I know this is copied from MM behavior but I'm wondering if we can do better since creating a new object of TopicPartition seem not a good pattern: instead of iterating over the whole `ConsumerRecords`, we can actually iterate per partition based on its `partitions()` and `List<ConsumerRecord<K, V>> records(TopicPartition partition)` api.